File: modulo1_logica/aula1_variaveis.cpp
// Aula 1: Variáveis e Matemática Básica
// Objetivo: Mostrar como o C++ funciona como uma calculadora super poderosa.

#include <iostream>
#include <string>

using namespace std;

int main() {
    cout << "--- AULA 1: BEM-VINDO AO C++! ---" << endl;
    cout << "Ola, futuro programador(a)!\n" << endl;

    // 1. Variáveis (Guardando informações na memória)
    string nome_aluno;
    double nota_bimestre1;
    double nota_bimestre2;

    // 2. Interação com o usuário (Entrada de Dados)
    cout << "Digite o nome do aluno: ";
    getline(cin, nome_aluno); // Usamos getline para permitir nomes com espaço

    cout << "Digite a nota do 1o bimestre: ";
    cin >> nota_bimestre1; // 'cin' lê o que o usuário digita no teclado

    cout << "Digite a nota do 2o bimestre: ";
    cin >> nota_bimestre2;

    cout << "\nCalculando as notas do aluno: " << nome_aluno << endl;

    // 3. Operações Matemáticas Simples (Lógica aplicada)
    // O computador faz a conta sozinho e guarda o resultado
    double media = (nota_bimestre1 + nota_bimestre2) / 2.0;

    cout << "A media final do aluno e: " << media << endl;

    // ==========================================
    // EXERCÍCIO PARA OS ALUNOS NA SALA DE AULA:
    // Execute o programa várias vezes testando notas diferentes
    // e perceba que agora o código é dinâmico e serve para qualquer aluno!
    // ==========================================

    return 0;
}

File: modulo1_logica/aula1_variaveis.py
# Aula 1: Nosso primeiro programa, Variáveis e Matemática Básica
# Objetivo: Mostrar como o Python funciona como uma calculadora super poderosa.

print("--- AULA 1: BEM-VINDO AO PYTHON! ---")
print("Olá, futuro programador(a)!")

# 1. Variáveis (Guardando informações na memória)
# Vamos imaginar que estamos calculando a nota de um aluno
nome_aluno = "Joãozinho"
nota_bimestre1 = 7.5
nota_bimestre2 = 8.0

print(f"\nCalculando as notas do aluno: {nome_aluno}")
print(f"Nota 1: {nota_bimestre1}")
print(f"Nota 2: {nota_bimestre2}")

# 2. Operações Matemáticas Simples (Lógica aplicada)
# O Python faz a conta sozinho e guarda em 'media'
media = (nota_bimestre1 + nota_bimestre2) / 2

print(f"A média final do aluno é: {media}")

# 3. Tomada de Decisão (If/Else) - A mágica acontece aqui!
print("\n--- RESULTADO FINAL ---")

if media >= 7.0:
    print("Situação: APROVADO! Parabéns, você foi muito bem.")
elif media >= 5.0:
    print("Situação: RECUPERAÇÃO. Estude um pouco mais.")
else:
    print("Situação: REPROVADO. Vamos tentar de novo ano que vem.")

# ==========================================
# EXERCÍCIO PARA OS ALUNOS NA SALA DE AULA:
# Peça para os alunos mudarem as variáveis 'nota_bimestre1' e 'nota_bimestre2'
# e rodarem o programa novamente para verem se o Joãozinho é aprovado ou não!
# ==========================================
